src/Microsoft.ServiceFabric.Actors/Runtime/MigrationActorStateProvider.cs

```diff
@@ -0,0 +1,445 @@
+
+        private async Task SaveRCData(List<RCData> rcDataList, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            using (var tx = this.rcStateProvider.GetStateManager().CreateTransaction())
+            {
+                foreach (RCData data in rcDataList)
+                {
+                    if (data.Value.Length > 0)
+                    {
+                        var dict = data.Dictionary;
+                        cancellationToken.ThrowIfCancellationRequested();
+                        await dict.AddOrUpdateAsync(tx, data.Key, data.Value, (k, v) => data.Value);
+                    }
+                }
+
+                await tx.CommitAsync();
+            }
+        }
+
+        private bool IsPresenceKey(string key)
+        {
+            string[] splitString = key.Split('_');
+            if (splitString.Length != 3 || splitString[0] != "@@")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool IsReminderCompletedKey(string key)
+        {
+            string[] splitString = key.Split('_');
+            if (splitString.Length != 4 || splitString[0] != "RC@@")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool IsLogicalTimeKey(string key)
+        {
+            string[] splitString = key.Split('_');
+            if (splitString.Length != 2 || splitString[1] != "VLTM")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool IsActorStateKey(string key)
+        {
+            string[] splitString = key.Split('_');
+            if (splitString.Length != 4 || splitString[0] != "Actor")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool IsReminderKey(string key)
+        {
+            string[] splitString = key.Split('_');
+            if (splitString.Length != 4 || splitString[0] != "Reminder")
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private ReminderCompletedData DeserializeReminderCompletedData(byte[] data)
+        {
+            DataContractSerializer serializer = new DataContractSerializer(typeof(ReminderCompletedData));
+            using (var reader = XmlDictionaryReader.CreateBinaryReader(data, XmlDictionaryReaderQuotas.Max))
+            {
+                return (ReminderCompletedData)serializer.ReadObject(reader);
+            }
+        }
+
+        private byte[] SerializeReminderCompletedData(ReminderCompletedData data)
+        {
+            return ReminderCompletedDataSerializer.Serialize(data);
+        }
+
+        private ActorReminderData DeserializeReminder(byte[] data)
+        {
+            DataContractSerializer serializer = new DataContractSerializer(typeof(ActorReminderData));
```

Comment:
Not sure if it helps improve performance. You can reuse the serializer object by making it an instance member or static member.

Reply:
Updated